.github/workflows/nightly.yml

```diff
@@ -119,7 +119,7 @@ jobs:
       timeout-minutes: 700
       run: |
         docker exec --workdir /src/MLOS/source/Mlos.Python mlos-build-ubuntu-${{ matrix.UbuntuVersion }} \
-          python3.7 -m unittest discover -p LongHaulTest*.py --verbose
+          pytest -sv mlos/Optimizers/long_haul_test/*
```

Comment:
Do we can which version of python that runs?

Reply:
Do you mean care? Well it's the one that installed pytest. Generally I'm highly skeptical about not using conda/virtualenv as it's really hard to control what is run. We might be able to run via python3.7 but then if that's not where we installed it earlier, it's not gonna work? Generally we should make sure that pytest and mlos where installed in the same environment, and then use that same environment.

Reply:
Agreed, but I think we'd tabled switching to conda for a future PR (that hasn't bubbled to the surface yet).
In the meantime, I've been trying to just explicitly reference python3.7 everywhere to avoid those types of confusions.
I think we can do that here with `python3.7 -m pytest` right?
Anyways, just a thought, I have no strong convictions on this atm given we recognize that some sort of virtual environment is the better long term approach (and as long as what we use works for now).

Reply:
Actually, given other comments, should this continue to run with `unittest` for now?

Reply:
I think this is fine, do you think there's an issue? I had thought about making this more elegant instead of using the particular file match, but I can live with this for now, and maybe we can later refactor the test.

Reply:
Yeah, it's fine.  In a previous comment I thought you'd said you didn't fixup the LongHaul tests to use pytest instead of unittest yet, but I think that was out of date.